components/feature/scroll/src/main/java/mozilla/components/feature/scroll/ScrollFeature.kt

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.scroll
+
+import android.support.design.widget.AppBarLayout
+import android.support.design.widget.AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL
+import android.support.design.widget.AppBarLayout.LayoutParams.SCROLL_FLAG_ENTER_ALWAYS
+import android.support.design.widget.AppBarLayout.LayoutParams.SCROLL_FLAG_SNAP
+import android.support.design.widget.AppBarLayout.LayoutParams.SCROLL_FLAG_EXIT_UNTIL_COLLAPSED
+import android.view.View
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.EngineView
+
+/**
+ * Feature implementation for connecting an [EngineView] with any View that you want to coordinate scrolling
+ * behavior with.
+ */
+class ScrollFeature(
```

Comment:
naming nit: `Scroll` is not precise enough yet, I think. Can we find something that is describing better was that feature does? I was thinking of something like "ToolbarHidingFeature" but its more generic than just about the toolbar (however we should mention the toolbar use case in the comment maybe!).

Reply:
CoordinateScrollingFeature? SyncScrollingFeature? ScrollFlagsFeature?

Reply:
Yeah name is hard 😅. I like `CoordinateScrollingFeature` 👍 .